### PR TITLE
EDGECLOUD-5537 EDGECLOUD-5563 EDGECLOUD-5414

### PR DESCRIPTION
**Pull request by brucearmstrong**
- Reduce the number of points used to draw the driving route. Processing too many points during animation was bogging down older phones. Added new "Reduction" preference to allow fine tuning.
- Fix crash when turning off route mode during playback.
- When new cloudlet is introduced via FindCloudletEvent, draw it in green to show it is closest.
- When cloudlet is removed, mark it as such, so we can change the info window shippet to "Has been removed".
- Hide "Verify Location" menu item as we don't currently want to expose that API call.
- Enable desugaring to support java.time.LocalDateTime from Java 8 on API < 26.

### CODE REVIEW

File: android/MobiledgeXSDKDemo/build.gradle [hunk 1]
```diff
@@ -1,13 +1,13 @@
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 
-project.ext.matchingengineVersion = "3.0-rc2-rebuild"
+project.ext.matchingengineVersion = "3.0-rc3"
```
Comment (ah1053) [inline android/MobiledgeXSDKDemo/build.gradle:3]: I don't if you need to update this to 3.0.0 or not, I think rc3 is 3.0.0

File: android/MobiledgeXSDKDemo/matchingenginehelper/src/main/java/com/mobiledgex/matchingenginehelper/MatchingEngineHelper.java [hunk 2]
```diff
@@ -1144,6 +1150,14 @@ public void postLocationToEdgeEvents(Location location) {
         }).start();
     }
 
+    public String getClosestCloudletHostname() {
+        if (mClosestCloudlet == null) {
+            return null;
+        } else {
+            return mClosestCloudlet.getFqdn();
+        }
+    }
```
Comment (ah1053) [inline android/MobiledgeXSDKDemo/matchingenginehelper/src/main/java/com/mobiledgex/matchingenginehelper/MatchingEngineHelper.java:1159]: I think MatchingEngine has a  getHost(fcReply, port) method, but  getFqdn() output should be the same plus the domain name.

File: android/MobiledgeXSDKDemo/README.md [hunk 3]
```diff
@@ -5,94 +5,17 @@
...
-
-Note that for Recognition, only a single face is supported at a time. If more than one face appears in the given image, only one of them will be detected and recognized.
-
-#### Pose Detection
-Pose Detection uses OpenPose on a GPU-enabled cloudlet to detect the pose of human bodies. Like the other activities, images from the camera are sent to the cloudlet for processing. Instead of rectangular coordinates, points representing the bones of the pose(s) are received and rendered.
-
-There is no Edge/Cloud comparison for this activity. A single cloudlet is used for image processing.
-
-Note that Pose Detection does not use the result of "Find Closest Cloudlet" because there are a limited number of GPU-enabled cloudlets available.
```
Comment (ah1053) [inline android/MobiledgeXSDKDemo/README.md:88]: Just curious why are you deleting this, this functionality is still in the app? 
Maybe a link to DevPortal if you think it is very descriptive.

**Review by ah1053 — APPROVED** [review]

File: android/MobiledgeXSDKDemo/build.gradle (hunk 1 above)
Comment (lgarner) [reply android/MobiledgeXSDKDemo/build.gradle:3]: It's actually 3.0 now, but basically exactly rc3.

**Review by lgarner — APPROVED** [review]
Ok.

File: android/MobiledgeXSDKDemo/README.md (hunk 3 above)
Comment (brucearmstrong) [reply android/MobiledgeXSDKDemo/README.md:88]: Vasanth created https://developers.mobiledgex.com/product/mobiledgex-demo from this README, and I have updated that doc with the new features, including Driving Route mode. At the bottom of this shortened README, there is a link to that doc after "For more details".

File: android/MobiledgeXSDKDemo/build.gradle (hunk 1 above)
Comment (brucearmstrong) [reply android/MobiledgeXSDKDemo/build.gradle]: Updated to "3.0".

**Pull request merged**